engine/src/test/java/com/datatorrent/stram/plan/physical/PhysicalPlanTest.java

```diff
@@ -466,23 +465,28 @@ public void testRepartitioningScaleUp()
   /**
    * Test partitioning of an input operator (no input port).
    * Cover aspects that are not part of generic operator test.
+   * Test scaling from one to multiple partitions with unifier when one partition remains unmodified.
    */
   @Test
   public void testInputOperatorPartitioning()
   {
     LogicalPlan dag = new LogicalPlan();
-    TestInputOperator<Object> o1 = dag.addOperator("o1", new TestInputOperator<>());
+    final TestInputOperator<Object> o1 = dag.addOperator("o1", new TestInputOperator<>());
+    GenericTestOperator o2 = dag.addOperator("o2", GenericTestOperator.class);
+    dag.addStream("o1.outport1", o1.output, o2.inport1);
+
     OperatorMeta o1Meta = dag.getMeta(o1);
     dag.setOperatorAttribute(o1, OperatorContext.STATS_LISTENERS, Arrays.asList(new StatsListener[]{new PartitioningTest.PartitionLoadWatch()}));
-    dag.setOperatorAttribute(o1, OperatorContext.PARTITIONER, new StatelessPartitioner<TestInputOperator<Object>>(2));
+    TestPartitioner<TestInputOperator<Object>> partitioner = new TestPartitioner<TestInputOperator<Object>>();
+    dag.setOperatorAttribute(o1, OperatorContext.PARTITIONER, partitioner);
 
     TestPlanContext ctx = new TestPlanContext();
     dag.setAttribute(OperatorContext.STORAGE_AGENT, ctx);
     PhysicalPlan plan = new PhysicalPlan(dag, ctx);
     Assert.assertEquals("number of containers", 2, plan.getContainers().size());
 
     List<PTOperator> o1Partitions = plan.getOperators(o1Meta);
-    Assert.assertEquals("partition instances " + o1Partitions, 2, o1Partitions.size());
+    Assert.assertEquals("partitions " + o1Partitions, 1, o1Partitions.size());
     PTOperator o1p1 = o1Partitions.get(0);
 
```

Comment:
o1p1 does not change and should not be redeployed. If o1p1 was a new partition (new object from partitioner) then it would be redeployed.